server/src/main/java/org/elasticsearch/cluster/routing/allocation/ExistingShardsAllocator.java

```diff
@@ -43,7 +44,14 @@ public interface ExistingShardsAllocator {
      * Called during a round of allocation after attempting to allocate all the primaries but before any replicas, allowing the allocator
      * to prepare for replica allocation.
      */
-    void afterPrimariesBeforeReplicas(RoutingAllocation allocation);
+    default void afterPrimariesBeforeReplicas(@SuppressWarnings("unused") RoutingAllocation allocation) {
+        assert false : "must be overridden";
+        throw new UnsupportedOperationException();
+    }
```

Comment:
This is temporary, will remove it in a followup after fixing up the dependent code elsewhere.